bug: mcp-go would panic for stdio tools if the tool outputs a really long line

## Description
mcp-go would panic for stdio tools if the tool outputs a really long line
## Code Sample

```go
// Minimum code snippet to reproduce the issue
// Remove if not applicable
```

## Logs or Error Messages

I got this panic stack
```text
panic: runtime error: slice bounds out of range [:8192] with capacity 4096

goroutine 38 [running]:
bufio.(*Reader).ReadSlice(0x1400028c300, 0xa)
        /opt/homebrew/Cellar/go/1.24.4/libexec/src/bufio/bufio.go:355 +0x34c
bufio.(*Reader).collectFragments(0x1400028c300, 0xa)
        /opt/homebrew/Cellar/go/1.24.4/libexec/src/bufio/bufio.go:455 +0x6c
bufio.(*Reader).ReadString(0x1400028c300, 0xa)
        /opt/homebrew/Cellar/go/1.24.4/libexec/src/bufio/bufio.go:502 +0x50
github.com/mark3labs/mcp-go/client/transport.(*Stdio).readResponses(0x1400040c0c0)
        /Users/izzy/go/pkg/mod/github.com/mark3labs/mcp-go@v0.32.0/client/transport/stdio.go:170 +0x7c
github.com/mark3labs/mcp-go/client/transport.(*Stdio).Start.func1()
        /Users/izzy/go/pkg/mod/github.com/mark3labs/mcp-go@v0.32.0/client/transport/stdio.go:79 +0x38
created by github.com/mark3labs/mcp-go/client/transport.(*Stdio).Start in goroutine 1
        /Users/izzy/go/pkg/mod/github.com/mark3labs/mcp-go@v0.32.0/client/transport/stdio.go:77 +0x104
panic: runtime error: slice bounds out of range [:8192] with capacity 4096```

## Environment

 - Go version (see `go.mod`): go 1.24.3
 - mcp-go version (see `go.mod`): 0.32.0

## Possible Solution

Perhaps don't use bufio.Reader as stdin

--- a/client/transport/stdio.go
+++ b/client/transport/stdio.go
@@ -27,7 +27,7 @@ type Stdio struct {
 	cmd            *exec.Cmd
 	cmdFunc        CommandFunc
 	stdin          io.WriteCloser
-	stdout         *bufio.Reader
+	stdout         *bufio.Scanner
 	stderr         io.ReadCloser
 	responses      map[string]chan *JSONRPCResponse
 	mu             sync.RWMutex
@@ -72,7 +72,7 @@ func WithCommandLogger(logger util.Logger) StdioOption {
 func NewIO(input io.Reader, output io.WriteCloser, logging io.ReadCloser) *Stdio {
 	return &Stdio{
 		stdin:  output,
-		stdout: bufio.NewReader(input),
+		stdout: bufio.NewScanner(input),
 		stderr: logging,
 
 		responses: make(map[string]chan *JSONRPCResponse),
@@ -180,7 +180,7 @@ func (c *Stdio) spawnCommand(ctx context.Context) error {
 	c.cmd = cmd
 	c.stdin = stdin
 	c.stderr = stderr
-	c.stdout = bufio.NewReader(stdout)
+	c.stdout = bufio.NewScanner(stdout)
 
 	if err := cmd.Start(); err != nil {
 		return fmt.Errorf("failed to start command: %w", err)
@@ -247,14 +247,15 @@ func (c *Stdio) readResponses() {
 		case <-c.done:
 			return
 		default:
-			line, err := c.stdout.ReadString('\n')
-			if err != nil {
-				if err != io.EOF && !errors.Is(err, context.Canceled) {
+			if !c.stdout.Scan() {
+				err := c.stdout.Err()
+				if err != nil && !errors.Is(err, context.Canceled) {
 					c.logger.Errorf("Error reading from stdout: %v", err)
 				}
 				return
 			}
 
+			line := c.stdout.Text()
 			// First try to parse as a generic message to check for ID field
 			var baseMessage struct {
 				JSONRPC string         `json:"jsonrpc"`

